feat: add status color tokens

marionnegp opened the pull request:
This PR:
- adds default, border, and surface status tokens
- fixes a duplicated red orange value
- changes the max width of the copy button to show full token name
- changes order of categories on Netlify page

To see what these tokens look like when used with our components: [Figma file](https://www.figma.com/file/FAhXoct6VdXsoRDkwHBLMK/Status-colors?type=design&node-id=28%3A562&mode=design&t=LfGcXsxzmPtsO0fF-1)

bennypowers commented on tokens/color/status.yaml:
```diff
@@ -0,0 +1,95 @@
+          $value: '{color.gray.80}'
+          $description: Neutral surface color (dark theme)
+
+    note:
```
these names don't exactly line up with the names already in use in `<rh-alert>`. are these names set in stone from on high?

https://ux.redhat.com/elements/alert/code/

coreyvickery replied:
@bennypowers Marionne and I are using `Alert` to see how we can start applying the new the colors to elements with statuses, so no, not from on high, but we are aiming for a hardened convention that can be token-ized across more elements.

marionnegp replied:
@bennypowers, I was using [this Figma file](https://www.figma.com/file/1YkKJofXZGfjHHt2t8ajWn/2023-color-palette?type=design&node-id=0-1&mode=design&t=qTwUV92lfGmomwhR-0) from Brand/DPO to name the different status levels. 

The only alert variant name that will have to change is "info," and "caution" and "neutral" will have to be added. Colors will have to change, but hopefully this will be an improvement because `<rh-alert>` is using crayon names right now.

bennypowers replied:
these are the alert states. changing them would break backwards


- default - Indicates generic information or a message with no severity.
- info - Indicates helpful information or a message with very little to no severity.
- success - Indicates a success state, like if a process was completed without errors.
- warning - Indicates a caution state, like a non-blocking error that might need to be fixed.
- danger - Indicates a danger state, like an error that is blocking a user from completing a task.

marionnegp replied:
Would we want to tackle `<rh-alert>` changes before merging status tokens then?

marionnegp replied:
We will likely have to update `<rh-badge>` as well. It uses the following states: info, success, moderate, important, critical.

bennypowers replied:
Alert has first mover advantage here, and imo it behooves us to carefully weigh that advantage.

Tokens can adopt alert's nomenclature without issue, because tokens as-of-yet have no concept of state.

Or token can innovate a new nomenclature. But then: either alert is retroactively rendered odd, or tokens are seen as overly abstract.

At that point there are two options (assuming tokens are authoritative), we can either break alert (and every instance of it, worldwide); or we can assert that alert has its own private language for state, then privately map alert states to token states

-----

I prefer for tokens to adopt alert states, because alert exists today whereas token states are still hypothetical

coreyvickery replied:
@bennypowers Excellent insight as always.

@marionnegp Does this help us move forward?

marionnegp replied:
@coreyvickery, yep! Thanks, @bennypowers. I also checked with Lucia from the PatternFly team, and they're planning on keeping their alert names the same. Since `<rh-alert>`'s names are already aligned with PatternFly's, I think we can go ahead and model the status token names after alert.

marionnegp commented:
@coreyvickery @bennypowers, the status tokens are aligned to the alert types now. I've removed the `caution` tokens because that alert level doesn't currently exist, and we can always add it when the need arises. I've also added `status` to the border and surface status token names. This is to avoid confusion with a name like `--rh-color-surface-default-on-light`. PTAL when you have a chance!

coreyvickery reviewed (approved):
Looks good.

marionnegp commented:
@bennypowers, could you take a look at these status tokens when you get a chance this week?

bennypowers commented on tokens/color/status.yaml:
```diff
@@ -0,0 +1,45 @@
+color:
+  status:
+    $extensions:
+      com.redhat.ux:
+        order: 500
+
+    danger:
+      on-light:
+        $value: '{color.red-orange.50}'
+        $description: Danger accent color (light theme)
```
this produces `--rh-color-status-danger-on-light`, but the changeset mentions `--rh-color-danger-on-light`. which one reflects your intentions, the changeset or the yaml record?

bennypowers commented on tokens/color/surface.yaml:
```diff
@@ -25,3 +25,40 @@ color:
       $value: '{color.gray.95}'
       $description: Primary surface (dark theme)
+
+    status:
```
should these be `--rh-color-status-surface-` instead of `--rh-color-surface-status-`?
